rfcs/055-prismic-api/README.md

```diff
@@ -0,0 +1,249 @@
+            "place": "",
+            "guide": {
+                ... // reuse the DigitalLocation model
+            }
+        }
+    ],
+    "contributors": [ // behind ?includes=contributors
+        {
+            ... // reuse the Contributor model
+        },
+        ...
+    ],
+    "image": {
+        "thumbnail": {
+            ... // reuse the DigitalLocation model
+        },
+        "caption": ""
+    },
+    "seasons": [
+        "",
+        ...
+    ],
+}
+```
+
+```http
+GET /exhibitions?query=foo&filters=bar&sort=baz
+{
+    "type": "ResultList",
+    "results": [
+        {
+            ... // as above
+        },
+        ...
+    ],
+    "pageSize": 100,
+    "totalResults": 1000,
+    "totalPages": 10,
+    "nextPage": "/exhibitions?query=foo&filters=bar&sort=baz?page=2",
+}
+```
+
+#### Filter
+
+- date
+- status
+- place
+- contributor
+- format (ie permanent, exhibition, installation)
+- season
+
+#### Sort
+
+- date
+- relevance
+
+### `/events`
+
+#### Example response
+
+```http
+GET /events/{id}
+{
+    "id": "",
+    "format": "",
+    "title": "",
+    "status": "",
+    "isOnline": false,
+    "availableOnline": false,
+    "audience": "",
+    "instantiations": [
+        {
+            "start": "", // ISO 8601 datetime
+            "end": "", // ISO 8601 datetime
+            "place": ""
+        }
+    ],
+    "contributors": [ // behind ?includes=contributors
+        {
+            ... // reuse the Contributor model
+        },
+        ...
+    ],
+    "image": {
+        "thumbnail": {
+            ... // reuse the DigitalLocation model
+        },
+        "caption": ""
+    },
+    "partOf": [
+        {
+            ... // reuse the partOf model for series and seasons
+        }
+    ]
+}
+```
+
+```http
+GET /events?query=foo&filters=bar&sort=baz
+{
+    "type": "ResultList",
+    "results": [
+        {
+            ... // as above
+        },
+        ...
+    ],
+    "pageSize": 100,
+    "totalResults": 1000,
+    "totalPages": 10,
+    "nextPage": "/events?query=foo&filters=bar&sort=baz?page=2",
+}
+```
+
+#### Filter
+
+- date
+- accessibility status (is it audio described, signed, etc)
+- location
+- format (eg discussion)
+- series
+- season
+- audience (eg schools)
+- contributor
+- isOnline?
+- availableOnline?
+
+#### Sort
+
+- date
+- relevance
+
+## Open questions
+
+- How do we represent functional content?
+- If we want to avoid maintaining multiple versions of the same code, should this work be done as an extension to the concepts API?
+- Works exist at [api.wellcomecollection.org/catalogue/v2/works](api.wellcomecollection.org/catalogue/v2/works), concepts at [/catalogue/v2/concepts](api.wellcomecollection.org/catalogue/v2/concepts), images at [/catalogue/v2/images](api.wellcomecollection.org/catalogue/v2/images).
+Stories etc aren't part of the catalogue - should the URL be different? Does `/content` work?
+- Should we include a `/series` endpoint as part of this work?
```

Comment:
@davidpmccormick it would be great if you could re-state the problem behind the "Rearranging comics on the stories landing page" toggle in this thread!

It sounds like a `/series` endpoint is something we should aim to include, but creating a matching pair of `article` and `series` es indices could be a tricky bit of work for the prismic pipeline folks. I don't know whether it's something for the first pass, or something which should be left until after the basics have been implemented. Probably a question for @jtweed to answer? maybe @jamieparkinson has thoughts?

Reply:
What is a `series`?

Reply:
What is a series?

Reply:
@dannybirchall might explain better but from I gathered, it's a grouping of articles/webcomics/other types. 
You'll find them under `/series/[id]` (https://wellcomecollection.org/series/WsSUrR8AAL3KGFPF) but there is no `/series` listing.
If they are scheduled (for every Thursday, for example), they are called a Serial, which is just a type of Series.

Reply:
With the 'Rearranging comics on the stories landing page' [toggle](https://dash.wellcomecollection.org/toggles/) turned on we add cards linking to the three most recent comic series on the stories landing page

<img width="1295" alt="Screenshot 2023-02-23 at 11 31 29" src="https://user-images.githubusercontent.com/1394592/220894243-4343bc84-bc6d-44e2-9afb-c0766b8d5408.png">

This is the [most recent comic series page](https://wellcomecollection.org/series/Y8aThBEAAKREABhL) for example.

These series don't have knowledge of the articles that are contained in them (the link is specified on the articles themselves) so when we query for them we don't know if they contain any articles yet. The series have to be published in order for the articles to link to them, so there will always be a point (however brief) when they don't have any articles attached (this isn't something that can be done as an inline in Prismic when e.g. publishing the first article).

Article series that don't contain any articles will currently 404.

At present we're asking the content team to add a `delist` tag to the series until the first article is published, but 1. maybe `delist`ed things will still show up in search(?) and 2. it isn't great to put the onus on the content team to remember to do this.

Reply:
> 1. maybe delisted things will still show up in search(?)

I can confirm they do, we allow them to show up in search after confirming with the content team. They use `delist` to curate older content before rendering it in `/articles`, but believe they should still be searchable.

Reply:
Agree that this is out of scope until we have search covered off. @jamieparkinson is also correct as to how to think about this - if we want a list of `x`, then we generate that by filtering `/content/v1/x`. This is consistent with how the catalogue api works and allow us to inherit pagination etc. Cf hierarchical works, ie archives and library series.

Reply:
Wondering about the `delist` issue and I’m thinking even if I go with a different approach that doesn’t involve requiring the content team to add that tag, the comic series will necessarily still have to be published before any comics are attached to it, so it will (presumably) still show up in search results?

Reply:
Resolving thread as it was addressed in [David's PR](https://github.com/wellcomecollection/wellcomecollection.org/pull/9362)